lua/nvim-tree/git/init.lua

```diff
@@ -134,6 +134,16 @@ function M.load_project_status(cwd)
   return M.projects[project_root]
 end
 
+function M.purge_state()
+  for _, p in pairs(M.projects) do
+    if p.watcher then
+      p.watcher:stop()
```

Comment:
It might be easier (and more reliable) to just create a `Watcher.purge` that goes through all `_watchers`.

We won't need to clear node and git watchers explicitly, and any future watchers will be purged with no effort required.

Reply:
for log.lua, i don't think we should clear M.path, since it could help detecting subsequent calls. Also the setup function of log.lua already removes the file if the truncate option is true.